ports/stm32/pin.c

```diff
@@ -601,6 +605,44 @@ MP_DEFINE_CONST_OBJ_TYPE(
     locals_dict, &pin_locals_dict
     );
 
+/******************************************************************************/
+// Special analog-only pin.
+
+#if MICROPY_HW_ENABLE_ANALOG_ONLY_PINS
+
+// When both normal and _C pins are defined in pins.csv, force the corresponding
+// analog switch to be open.  The macro code below will produce a compiler warning
+// if the board defines the switch as closed.
+#if defined(pin_A0) && defined(pin_A0_C)
+#define MICROPY_HW_ANALOG_SWITCH_PA0 (SYSCFG_SWITCH_PA0_OPEN)
```

Comment:
Is the idea that this triggers a redefinition of the macro?

Would doing this achieve the same:

```c
#if defined(pin_A0) && defined(pin_A0_C) && defined(MICROPY_HW_ANALOG_SWITCH_PA0) && MICROPY_HW_ANALOG_SWITCH_PA0 == SYSCFG_SWITCH_PA0_CLOSED
#error "Both pads for dual-pad A0 are enabled, switch must be closed."
#endif
```

(It's a bit confusing to see a .c file defining a macro that it doesn't use)

Reply:
As the comment says, this is intended to trigger a compiler warning if the board defines the switch as closed.

I originally had tried what you suggested above but it doesn't work because the comparison `MICROPY_HW_ANALOG_SWITCH_PA0 == SYSCFG_SWITCH_PA0_CLOSED` is too complex for the preprocessor.